libraries/config/src/config.cpp

```diff
@@ -80,20 +73,37 @@ FullNodeConfig::FullNodeConfig(const Json::Value &string_or_object, const Json::
       }
     }
   }
+
+  is_light_node = getConfigDataAsBoolean(root, {"is_light_node"}, true, is_light_node);
+  light_node_history = getConfigDataAsUInt(root, {"light_node_history"}, true, light_node_history);
+}
+
+FullNodeConfig::FullNodeConfig(const Json::Value &string_or_object, const Json::Value &wallet,
+                               const Json::Value &genesis_json, const std::string &config_file_path) {
+  Json::Value parsed_from_file = getJsonFromFileOrString(string_or_object);
+  if (string_or_object.isString()) {
+    json_file_name = string_or_object.asString();
+  } else {
+    json_file_name = config_file_path;
+  }
+  assert(!json_file_name.empty());
+
+  auto const &root = string_or_object.isString() ? parsed_from_file : string_or_object;
+  overwriteConfigFromJson(root);
+
   if (const auto &v = genesis_json; v.isObject()) {
     dec_json(v, genesis);
   } else {
-    genesis = Genesis::predefined();
+    genesis = Genesis();
   }
-
   // blocks_per_year config param is calculated from lambda_ms
   uint64_t year_ms = 365 * 24 * 60 * 60;
   year_ms *= 1000;
   // we have fixed 2*lambda time for proposing step and adding some expecting value for filter and certify steps
   const uint32_t expected_block_time = 2 * genesis.pbft.lambda_ms + 700;
   genesis.state.dpos.blocks_per_year = year_ms / expected_block_time;
 
-  is_light_node = getConfigDataAsBoolean(root, {"is_light_node"}, true, false);
+  is_light_node = getConfigDataAsBoolean(root, {"is_light_node"}, true, is_light_node);
```

Comment:
hmm last argument == default value, why is it changed from false to is_light_node ?

Reply:
It basically changing nothing, just seems to be more logical for me to keep previous value if don't have value to set